addons/aircraft/functions/fnc_canShowEject.sqf

```diff
@@ -0,0 +1,36 @@
+/*
+ * Author: Dystopian
+ * Check if Eject action can be shown.
+ *
+ * Arguments:
+ * 0: Unit who calls action <OBJECT>
+ * 1: Vehicle which action is attached to <OBJECT>
+ *
+ * Return Value:
+ * Can show <BOOL>
+ *
+ * Example:
+ * [player, vehicle player] call ace_aircraft_fnc_canShowEject
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+params [["_unit", objNull, [objNull]], ["_vehicle", objNull, [objNull]]];
+
+_vehicle == vehicle _unit
+&& {_vehicle getVariable [
+    // handle race when unit gets out of vehicle
+    call {
+        private _return = {
+            _x params ["_owner", "_role", "", "_turretPath"];
+            if (_unit == _owner) exitWith {
+                format ["%1_%2_%3", QGVAR(ejectAction), _role, _turretPath]
+            };
+            false
+        } count fullCrew _vehicle;
+        // getVariable ["", ...] returns default value
+        ["", _return] select (_return isEqualType "")
+    },
+    false
```

Comment:
My brain hurts.

Reply:
Performance, not Readability, sorry. All I can do for this implementation is to add more comments and/or correct indents.

Reply:
You could get rid of the type validation of the first params for performance.

Reply:
```sqf
#include "script_component.hpp"

params ["_unit", "_vehicle"];

_vehicle == vehicle _unit && {
    private _return = "";

    {
        _x params ["_owner", "_role", "", "_turretPath"];

        if (_unit isEqualTo _owner) exitWith {
            _return = format ["%1_%2_%3", QGVAR(ejectAction), _role, _turretPath]
        };
    } count fullCrew _vehicle;

    _vehicle getVariable [_return, false] // return
};
```
???

Reply:
Btw, shouldn't the turret path already make this unique? Do we need the role at all?

Reply:
Your code is faster and more readable, thank you.
The role is deleted too.

Reply:
The role was to distinguish driver and cargo man (both have `[]` turret). I have to restore it.